docs/examples/multi_gate.rst

```diff
@@ -88,269 +90,509 @@ To run the code in this example, the following is required.
 Formulating the Problem as a CSP
 ================================
 
-This example demonstrates two formulations of constraints from the problem's
-logic gates:
+Other examples (:ref:`not` and :ref:`and`) show how a Boolean gate is
+represented as a *constraint satisfaction problem* (:term:`CSP`) on a quantum
+computer. This example aggregates the resulting binary quadratic models (BQM)
+to do the same for multiple gates that constitute a circuit.
 
-1. Single comprehensive constraint.
+Small-Circuit Problem
+---------------------
 
-.. testcode::
+To start with, solve the circuit with 7 logic gates shown above.
 
-    import dwavebinarycsp
+The code below represents standard Boolean gates with BQMs provided by
+:doc:`dimod </docs_dimod/sdk_index>` BQM generators, represents the Boolean NOT
+operation by inverting the coefficients of the relevant variables, and sums the
+gate-level BQMs. The resulting aggregate BQM has its lowest value for variable
+assignments that satisfy all the constraints representing the circuit's Boolean gates.
 
-    def logic_circuit(a, b, c, d, z):
-        not1 = not b
-        or2 = b or c
-        and3 = a and not1
-        or4 = or2 or d
-        and5 = and3 and or4
-        not6 = not or4
-        or7 = and5 or not6
-        return (z == or7)
-
-    csp = dwavebinarycsp.ConstraintSatisfactionProblem(dwavebinarycsp.BINARY)
-    csp.add_constraint(logic_circuit, ['a', 'b', 'c', 'd', 'z'])
-
-
-2. Multiple small constraints.
-
-.. testcode::
-
-    import dwavebinarycsp
-    import dwavebinarycsp.factories.constraint.gates as gates
-    import operator
-
-    csp = dwavebinarycsp.ConstraintSatisfactionProblem(dwavebinarycsp.BINARY)
-    csp.add_constraint(operator.ne, ['b', 'not1'])  # add NOT 1 gate
-    csp.add_constraint(gates.or_gate(['b', 'c', 'or2']))  # add OR 2 gate
-    csp.add_constraint(gates.and_gate(['a', 'not1', 'and3']))  # add AND 3 gate
-    csp.add_constraint(gates.or_gate(['d', 'or2', 'or4']))  # add OR 4 gate
-    csp.add_constraint(gates.and_gate(['and3', 'or4', 'and5']))  # add AND 5 gate
-    csp.add_constraint(operator.ne, ['or4', 'not6'])  # add NOT 6 gate
-    csp.add_constraint(gates.or_gate(['and5', 'not6', 'z']))  # add OR 7 gate
-
-.. note:: `dwavebinarycsp` works best for constraints of up to 4 variables; it may not
-      function as expected for constraints of over 8 variables.
-
-The next line of code converts the constraints into a BQM that you solve by sampling.
-
-.. testcode::
+>>> from dimod.generators import and_gate, or_gate
+...
+>>> bqm2 = or_gate('b', 'c', 'out2')
```

Comment:
why not name these functionally? Like `or_bc = or_gate('b', 'c', 'out2')` or whatever

Reply:
It makes an easy connection between the BQM for a gate and that gate's numbering in the figure above it (e.g., the users sees "bqm4" and then looks up the gate with a big "4" on it in the figure) 
The functionality is easily readable from the rhs (operation in the gate name and inputs in the arguments) so adding those to the lhs wouldn't add helpful info. 
If this content moves in the future somewhere that separates code from description (e.g., an example with a readme or JN) then functional names would be better, but close on the same page, this makes a nice connection

Reply:
Sure, defer to what you think makes sense. Though IMO if that's the reasoning I might name them `gate4 = or_gate(...)`

Reply:
Sure, I updated to ``bqm_gate<num gate>`` (I want to also keep the "bqm" to emphasize that they are creating BQMs and then adding them to a comprehensive BQM